build fails: Could not download Node.js: Got error code 404 from the server.

OS: Mac OS
CPU: M1

It is trying to download old version of node that is not released for M1 
`https://nodejs.org/dist/v14.16.0/node-v14.16.0-darwin-arm64.tar.gz`

```bash
~/repo/spline-ui % mvn install
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] project                                                            [pom]
[INFO] UI Core                                                            [pom]
[INFO] Spline Web UI                                                      [war]
[INFO]
[INFO] --------------------< za.co.absa.spline.ui:project >--------------------
[INFO] Building project 1.0.0-SNAPSHOT                                    [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- maven-install-plugin:2.4:install (default-install) @ project ---
[INFO] Installing /Users/abac720/repo/spline-ui/pom.xml to /Users/abac720/.m2/repository/za/co/absa/spline/ui/project/1.0.0-SNAPSHOT/project-1.0.0-SNAPSHOT.pom
[INFO]
[INFO] --------------------< za.co.absa.spline.ui:ui-core >--------------------
[INFO] Building UI Core 1.0.0-SNAPSHOT                                    [2/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- frontend-maven-plugin:1.10.3:install-node-and-npm (install node and npm) @ ui-core ---
[INFO] Installing node version v14.16.0
[INFO] Downloading https://nodejs.org/dist/v14.16.0/node-v14.16.0-darwin-arm64.tar.gz to /Users/abac720/.m2/repository/com/github/eirslett/node/14.16.0/node-14.16.0-darwin-arm64.tar.gz
[INFO] No proxies configured
[INFO] No proxy was configured, downloading directly
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for project 1.0.0-SNAPSHOT:
[INFO]
[INFO] project ............................................ SUCCESS [  0.117 s]
[INFO] UI Core ............................................ FAILURE [  0.509 s]
[INFO] Spline Web UI ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.686 s
[INFO] Finished at: 2022-11-29T15:07:23+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.10.3:install-node-and-npm (install node and npm) on project ui-core: Could not download Node.js: Got error code 404 from the server. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :ui-core
```

I tried to use new version of node, but then I got another error:

```bash
[INFO] --- frontend-maven-plugin:1.10.3:npm (install node modules) @ ui-core ---
[INFO] Running 'npm ci' in /Users/abac720/repo/spline-ui/ui
[INFO] npm ERR! code ERESOLVE
[INFO] npm ERR! ERESOLVE could not resolve
[INFO] npm ERR!
[INFO] npm ERR! While resolving: @angular-eslint/builder@0.0.1-alpha.27
[INFO] npm ERR! Found: @angular-devkit/build-angular@12.1.3
[INFO] npm ERR! node_modules/@angular-devkit/build-angular
[INFO] npm ERR!   dev @angular-devkit/build-angular@"12.1.3" from the root project
[INFO] npm ERR!   @angular-devkit/build-angular@"^12.0.0" from @angular-builders/custom-webpack@12.1.0
[INFO] npm ERR!   node_modules/@angular-builders/custom-webpack
[INFO] npm ERR!     dev @angular-builders/custom-webpack@"12.1.0" from the root project
[INFO] npm ERR!
[INFO] npm ERR! Could not resolve dependency:
[INFO] npm ERR! peer @angular-devkit/build-angular@"~0.900.2" from @angular-eslint/builder@0.0.1-alpha.27
[INFO] npm ERR! node_modules/@angular-eslint/builder
[INFO] npm ERR!   dev @angular-eslint/builder@"0.0.1-alpha.27" from the root project
[INFO] npm ERR!
[INFO] npm ERR! Conflicting peer dependency: @angular-devkit/build-angular@0.900.7
[INFO] npm ERR! node_modules/@angular-devkit/build-angular
[INFO] npm ERR!   peer @angular-devkit/build-angular@"~0.900.2" from @angular-eslint/builder@0.0.1-alpha.27
[INFO] npm ERR!   node_modules/@angular-eslint/builder
[INFO] npm ERR!     dev @angular-eslint/builder@"0.0.1-alpha.27" from the root project
[INFO] npm ERR!
[INFO] npm ERR! Fix the upstream dependency conflict, or retry
[INFO] npm ERR! this command with --force, or --legacy-peer-deps
[INFO] npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
[INFO] npm ERR!
[INFO] npm ERR! See /Users/abac720/.npm/eresolve-report.txt for a full report.
[INFO]
[INFO] npm ERR! A complete log of this run can be found in:
[INFO] npm ERR!     /Users/abac720/.npm/_logs/2022-11-29T14_54_27_471Z-debug-0.log
```

--- a/ui/pom.xml
+++ b/ui/pom.xml
@@ -76,7 +76,7 @@
                 <artifactId>frontend-maven-plugin</artifactId>
                 <version>1.10.3</version>
                 <configuration>
-                    <nodeVersion>v14.16.0</nodeVersion>
+                    <nodeVersion>v16.9.1</nodeVersion>
                     <npmVersion>7.6.0</npmVersion>
                     <environmentVariables>
                         <SPLINE_UI_BUILD_REVISION>${buildNumber}</SPLINE_UI_BUILD_REVISION>

